Dockerfile

```diff
@@ -1,5 +1,16 @@
-FROM adoptopenjdk/openjdk11:alpine-jre
-ADD https://github.com/microsoft/ApplicationInsights-Java/releases/download/3.2.6/applicationinsights-agent-3.2.6.jar /applicationinsights-agent.jar
+FROM maven:3.8.4-jdk-11-slim as buildtime
+
+WORKDIR /build
+COPY . .
+
+RUN mvn clean package
+
+FROM adoptopenjdk/openjdk11:alpine-jre as runtime
+
 VOLUME /tmp
-COPY target/*.jar app.jar
-ENTRYPOINT ["java","-jar","app.jar"]
\ No newline at end of file
+WORKDIR /app
+
+COPY --from=buildtime /build/target/*.jar /app/app.jar
+ADD https://github.com/microsoft/ApplicationInsights-Java/releases/download/3.2.7/applicationinsights-agent-3.2.7.jar /app/applicationinsights-agent.jar
+
```

Comment:
I'll add a comment to explain that we are setting the app insight agent via  `JAVA_TOOL_OPTIONS`

Reply:
Agree, could save some time in the future